ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/uplinks.py

```diff
@@ -184,148 +184,239 @@ def uplinks(self: EosDesignsFacts) -> list:
+                uplink["peer_ip_address"] = self.shared_utils.ip_addressing.p2p_uplinks_peer_ip(uplink_index)
 
-                uplinks.append(uplink)
-            return uplinks
+            if self.shared_utils.link_tracking_groups is not None:
+                uplink["link_tracking_groups"] = [{"name": lt_group["name"], "direction": "upstream"} for lt_group in self.shared_utils.link_tracking_groups]
+            if self.shared_utils.uplink_structured_config is not None:
+                uplink["structured_config"] = self.shared_utils.uplink_structured_config
 
-        elif self.shared_utils.uplink_type == "port-channel":
-            uplink_interfaces = self._uplink_interfaces
-            uplink_switches = self.shared_utils.uplink_switches
-            uplink_switch_interfaces = self._uplink_switch_interfaces
-            for uplink_index, uplink_interface in enumerate(uplink_interfaces):
-                if len(uplink_switches) <= uplink_index or len(uplink_switch_interfaces) <= uplink_index:
-                    # Invalid length of input variables. Skipping
-                    continue
-
-                uplink_switch = uplink_switches[uplink_index]
-                if uplink_switch is None or uplink_switch not in self.shared_utils.all_fabric_devices:
-                    # Invalid uplink_switch. Skipping.
-                    continue
-
-                uplink_switch_facts: EosDesignsFacts = self.shared_utils.get_peer_facts(uplink_switch, required=True)
-                uplink = {}
-                uplink["interface"] = uplink_interface
-                uplink["peer"] = uplink_switch
-                uplink["peer_interface"] = uplink_switch_interfaces[uplink_index]
-                uplink["peer_type"] = uplink_switch_facts.type
-                uplink["peer_is_deployed"] = uplink_switch_facts.is_deployed
-                uplink["type"] = "underlay_l2"
-
-                if self.shared_utils.uplink_interface_speed is not None:
-                    uplink["speed"] = self.shared_utils.uplink_interface_speed
-
-                if self.shared_utils.uplink_switch_interface_speed is not None:
-                    uplink["peer_speed"] = self.shared_utils.uplink_switch_interface_speed
-
-                if self.shared_utils.uplink_ptp is not None:
-                    uplink["ptp"] = self.shared_utils.uplink_ptp
-                elif self.shared_utils.ptp_enabled:
-                    uplink["ptp"] = {"enable": True}
-
-                if uplink_switch_facts.shared_utils.mlag is True or self._short_esi is not None:
-                    # Override our description on port-channel to be peer's group name if they are mlag pair or A/A #}
-                    uplink["channel_description"] = uplink_switch_facts.shared_utils.group
+            uplinks.append(uplink)
+        return uplinks
+
+    def _get_port_channel_uplinks(self: EosDesignsFacts) -> list:
```

Comment:
Hard to review these. Have you changed anything or just moved it around?

Reply:
moved it around